packages/clay-date-picker/src/index.tsx

```diff
@@ -284,10 +312,40 @@ const ClayDatePicker: React.FunctionComponent<IProps> = React.forwardRef<
 		/**
 		 * Handles the click on element of the day
 		 */
-		const handleDayClicked = (date: Date) => {
-			setDaySelected(date);
+		const handleDayClicked = (date: Date, enableDateRange?: boolean) => {
+			let newDateRange = dateRange;
+
+			if (enableDateRange) {
+				if (date < dateRange.fromDate) {
+					newDateRange = {...dateRange, fromDate: date};
+
+					setDateRange(newDateRange);
+
+					onValueChange(
+						fromDateRangeToString(newDateRange, dateFormat),
+						'click'
+					);
+
+					return;
+				}
+
+				newDateRange = {...newDateRange, toDate: date};
 
-			onValueChange(date, 'click');
+				setDateRange(newDateRange);
+
+				onValueChange(
+					fromDateRangeToString(newDateRange, dateFormat),
+					'click'
+				);
+
+				return;
+			}
+
+			newDateRange = {fromDate: date, toDate: date};
+
+			setDateRange(newDateRange);
+
+			onValueChange(formatDate(date, dateFormat), 'click');
```

Comment:
Just leaving it here for you to remember once we talked about it... but this can be improved both to avoid some lines of code and to improve the readability of the use cases that exist for the range behavior.

Reply:
I splitted this whole function into two parts :)

One for handling when having a range and the other for the default case